PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,261 @@
+    windowNames = driver.window_handles  # Get all window names
+    for name in windowNames:
+        if mainWindowName == name:
+            continue  # Do not switch to the main window because the search page opens from the main window
+
+        driver.switch_to_window(name)
+        break  # Because there are only two windows: the main window and the open window, switching directly to the end is over
+    # Waiting for elements in the page to appear, indicating that the page has finished loading
+    waitUtilPageLoaded(driver)
+
+
+def findMenu(driver):
+    """Find menu."""
+    menus = driver.find_elements_by_class_name("list-group-item")
+    if not menus:
+        print("No left menu found！")
+        return
+    appMenu = []
+    for menu in menus:
+        txt = menu.text
+        if ("apps".lower() == txt.lower()):  # No left menu found
+            appMenu = menu
+            break
+    if not appMenu:
+        print("Apps menu not found in left menu！")
+        return
+    time.sleep(2)  # Wait for the next page to come over
+    appMenu.click()  # Open the menu
+
+def findAppList(driver):
+    """Find app list."""
+    clss = driver.find_elements_by_class_name("myapps-item")
+    if not clss:
+        print("No app list found！")
+        return
+    classes = []
+    for cls in clss:
+        txt = cls.find_element_by_class_name("myapps-item-label").text
+        txt = str.strip(txt)
+        # print("text:  ",  txt)
+        if ("Look Up Classes".lower() == txt.lower()):
+            classes = cls
+            break
+    if not classes:
+        print("No Look Up Classes feature found in the app list!")
+        return
+
+
+# Go to the advanced options page and start filling in various search terms
+def fillAdvanceSearch(driver):
+    """Fill advance search."""
+    # Select all options in Subject list
+    subjectList = driver.find_element_by_id("subj_id")  # web element
+    subjectOptions = subjectList.find_elements_by_tag_name("option")  # list
+    subjectListSelect = Select(subjectList)
+    print("Start to select all the contents in the multi-selection drop-down box. There are many options. You may have to wait for more time. Don't worry ...")
+    for i in range(0, len(subjectOptions)):
+        subjectListSelect.select_by_index(i)
+    # Submit search and click
+    submitD(driver)
+
+
+def submitD(driver):
+    """Locate submit button and submit."""
+    ipts = driver.find_elements_by_tag_name("input")  # list of Web Element
+    submit = ""  # webElement
+    for ipt in ipts:
+        if "submit" == ipt.get_attribute("type").lower() and \
+                "Section Search".lower() == ipt.get_attribute("value").lower():
+            submit = ipt
+            break
+
+    if not submit:
+        print("No submit button found! ! ")
+        sys.exit(-1)  # Program exit
```

Comment:
or raise exception with error message

Reply:
But if the submit button was not found, the program has to exit.

Reply:
You may raise an exception here, and do `system.exit(-1)` in the exception handler

Reply:
If the submit button can not be found, then the program should be stop not only raise an exception.

Reply:
If an exception is raised here, the `try:...` `except:...` clause in the outer function will catch this exception and handles the exception using the code provided in the `except` clause (In this case, is doing `sys.exit(<some code>)`)